libcontainer/standard_init_linux.go

```diff
@@ -79,12 +66,41 @@ func (l *linuxStandardInit) Init() error {
 	}
 
 	label.Init()
-	// InitializeMountNamespace() can be executed only for a new mount namespace
+
+	// prepareRootfs() can be executed only for a new mount namespace
+	if l.config.Config.Namespaces.Contains(configs.NEWNS) {
+		if err := prepareRootfs(l.config.Config); err != nil {
+			return err
+		}
+	}
+
+	// Signal the parent to run the pre-start hooks.
+	// The hooks are run after the mounts are setup, but before we switch to the new
+	// root, so that the old root is still available in the hooks for any mount
+	// manipulations.
+	if err := syncParentHooks(l.pipe); err != nil {
```

Comment:
I just tried to implement the split with `PivotRoot` in `finalizeRootfs` and the code gets quite ugly quite quickly. So I'm just going to move the `syncParentHooks` back to `prepareRootfs` as it was before. As an aside, we really need to add an integration test for this behaviour of the hooks API.